src/mapd-con-es6.js

```diff
@@ -618,6 +618,294 @@ class MapdCon {
     )
   }
 
+  getUsersAsync = () =>
+    new Promise((resolve, reject) => {
+      if (this._sessionId) {
+        this._client[0].get_users(this._sessionId[0], result => {
+          if (result instanceof Error) {
+            reject(result)
+          } else {
+            resolve(result)
+          }
+        })
+      } else {
+        reject(
+          new Error(
+            "You are not connected to a server. Try running the connect method first."
+          )
+        )
+      }
+    })
+
+  getRolesAsync = () =>
+    new Promise((resolve, reject) => {
+      if (this._sessionId) {
+        this._client[0].get_roles(this._sessionId[0], result => {
+          if (result instanceof Error) {
+            reject(result)
+          } else {
+            resolve(result)
+          }
+        })
+      } else {
+        reject(
+          new Error(
+            "You are not connected to a server. Try running the connect method first."
+          )
+        )
+      }
+    })
+
+  getDashboardsAsync = () =>
+    new Promise((resolve, reject) => {
+      if (this._sessionId) {
+        this._client[0].get_dashboards(this._sessionId[0], result => {
+          if (result instanceof Error) {
+            reject(result)
+          } else {
+            resolve(result)
+          }
+        })
+      } else {
+        reject(
+          new Error(
+            "You are not connected to a server. Try running the connect method first."
+          )
+        )
+      }
+    })
+
+  getDashboardAsync = dashboardId =>
+    new Promise((resolve, reject) => {
+      if (this._sessionId) {
+        this._client[0].get_dashboard(
+          this._sessionId[0],
+          dashboardId,
+          result => {
+            if (result instanceof Error) {
+              reject(result)
+            } else {
+              resolve(result)
+            }
+          }
+        )
+      } else {
+        reject(
+          new Error(
+            "You are not connected to a server. Try running the connect method first."
+          )
+        )
+      }
+    })
+
+  /**
+   * Add a new dashboard to the server.
+   * @param {String} dashboardName - the name of the new dashboard
+   * @param {String} dashboardState - the base64-encoded state string of the new dashboard
+   * @param {String} imageHash - the numeric hash of the dashboard thumbnail
```

Comment:
This functionality of taking a screenshot of the dashboard would be cool.

Reply:
It would! Not sure if we have an issue for that yet?